src/frontend/js/utils/react-query/createQueryClient.ts

```diff
@@ -0,0 +1,50 @@
+import { createSessionStoragePersistor } from 'utils/react-query/createSessionStoragePersistor';
+import { persistQueryClient } from 'react-query/persistQueryClient-experimental';
+import { QueryClient, setLogger } from 'react-query';
+import { CommonDataProps } from 'types/commonDataProps';
+import { handle } from 'utils/errors/handle';
+import { REACT_QUERY_SETTINGS } from 'settings';
+import { noop } from 'utils';
+
+interface QueryClientOptions {
+  logger?: boolean;
+  persistor?: boolean;
+}
+
+const createQueryClient = (options?: QueryClientOptions) => {
+  const context: CommonDataProps['context'] = (window as any)?.__richie_frontend_context__?.context;
+  const environment = context?.environment;
+
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        cacheTime: REACT_QUERY_SETTINGS.cacheTime,
+        refetchOnWindowFocus: false,
```

Comment:
Is there a reason we want this disabled? In general I think it's a great feature of react-query.

Is it because it might cause inconsistencies with stale server-generated content (and we'd rather have the whole page be stale)?

Reply:
From my point of view, default react-query parameters are too agressive. Especially this option to refetch data each time user get focus on the window.

So I choose to disable it by default but I enable this option to retrieve session state. In this way, if the user logged out from OpenEdX, we will be able to destroy the user session from richie.